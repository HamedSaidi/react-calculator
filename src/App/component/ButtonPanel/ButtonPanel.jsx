import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';

import './ButtonPanel.css';

export default class ButtonPanel extends React.Component {
  static propTypes = {
      clickHandler: PropTypes.func.isRequired,
  }

  handleClick = (buttonName) => {
      const { clickHandler } = this.props;

      clickHandler(buttonName);
  }

  render() {
      const buttonsMatrice = [
          ['7', '4', '1', 'AC'],
          ['8', '5', '2', '0'],
          ['9', '6', '3', '.'],
          ['+', '-', '='],
      ];

      return (
          <div className="component-button-panel">
              {
                  buttonsMatrice.map(line => (
                      <div key={line[0]}>
                          {
                              line.map(name => (
                                  <Button
                                      name={name}
                                      key={name}
                                      clickHandler={this.handleClick}
                                  />
                              ))
                          }
                      </div>
                  ))
              }
          </div>
      );
  }
}
