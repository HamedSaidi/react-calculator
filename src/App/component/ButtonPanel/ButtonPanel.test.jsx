import React from 'react';
import { shallow } from 'enzyme';

import ButtonPanel from '.';

describe('ButtonPanel', () => {
    let consoleSpy;

    beforeAll(() => {
        consoleSpy = jest.spyOn(console, 'error');
    });

    it('renders the ButtonPanel component and matches snapshot', () => {
        const component = shallow(<ButtonPanel clickHandler={() => {}} />);

        expect(component).toMatchSnapshot();
    });

    it('shows no errors on console', () => {
        expect(consoleSpy).not.toHaveBeenCalled();
    });
});
