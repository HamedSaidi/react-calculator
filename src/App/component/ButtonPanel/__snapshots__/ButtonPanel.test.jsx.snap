// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel renders the ButtonPanel component and matches snapshot 1`] = `
<div
  className="component-button-panel"
>
  <div
    key="7"
  >
    <Button
      clickHandler={[Function]}
      key="7"
      name="7"
    />
    <Button
      clickHandler={[Function]}
      key="4"
      name="4"
    />
    <Button
      clickHandler={[Function]}
      key="1"
      name="1"
    />
    <Button
      clickHandler={[Function]}
      key="AC"
      name="AC"
    />
  </div>
  <div
    key="8"
  >
    <Button
      clickHandler={[Function]}
      key="8"
      name="8"
    />
    <Button
      clickHandler={[Function]}
      key="5"
      name="5"
    />
    <Button
      clickHandler={[Function]}
      key="2"
      name="2"
    />
    <Button
      clickHandler={[Function]}
      key="0"
      name="0"
    />
  </div>
  <div
    key="9"
  >
    <Button
      clickHandler={[Function]}
      key="9"
      name="9"
    />
    <Button
      clickHandler={[Function]}
      key="6"
      name="6"
    />
    <Button
      clickHandler={[Function]}
      key="3"
      name="3"
    />
    <Button
      clickHandler={[Function]}
      key="."
      name="."
    />
  </div>
  <div
    key="+"
  >
    <Button
      clickHandler={[Function]}
      key="+"
      name="+"
    />
    <Button
      clickHandler={[Function]}
      key="-"
      name="-"
    />
    <Button
      clickHandler={[Function]}
      key="="
      name="="
    />
  </div>
</div>
`;
